osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -814,6 +814,9 @@ private void ensurePlayingSelected()
             if (!ControlGlobalMusic)
                 return;
 
+            if (Beatmap.Value is DummyWorkingBeatmap)
+                return;
```

Comment:
I'm not sure where the failure vector is here or what this is trying to do exactly. Mind explaining this further? Is this something to do with the beatmap/ruleset debounce in song select?

Reply:
If the current beatmap is dummy (no selection), the global track can potentially still be the previous track.

In this code song select was telling a potentially arbitrary global track to play regardless of checking whether a selection is current. Then when returning to menu the same thing would happen there: